### DISCUSSION TITLE
Do I need to do anything if I turn on `sendPingsAutomatically`?

### DISCUSSION THREAD

**Discussion by vedantroy** [uNetworking/uWebSockets#1461] [Q&A]
If I turn on `sendPingsAutomatically`, in order to keep a connection alive, do I need to implement the client sending some sort of pong message as a response? Or is it sufficient to just set this option to `true` in order to keep connections alive?

**Comment by ghost** [comment]
Nothing is needed, that option is even default ;)

**Reply by matiaslopezd** [reply]
Do you know how to check the ping/pong on Wireshark? I try to search with:

PING
`rpcap.opcode == 0x9`

PONG
`rpcap.opcode == 0xA`

We're using proxies products like Cloudflare (as a proxy for production and tunnel for development) and for the tunnel in the development environment, we're experimenting with Websocket disconnection after a ~minute inclusive if the `idleTimeout` is 8. The default TCP timeout in Cloudflare is 120 and 270s.

@uNetworkingAB

**Reply by matiaslopezd** [reply]
Also, do you consider it a good idea to send PING/PONG manually, thinking in a SaaS enterprise app could work through firewalls? I mean the 4 bits opcode mentioned in RFC 6455, will be considered as data in a TCP connection in most of the firewalls?

@uNetworkingAB
